os_migrate/plugins/module_utils/server_port.py

```diff
@@ -78,7 +80,12 @@ def from_sdk(cls, conn, sdk_resource):
         return obj
 
     def create_or_update(self, conn, filters=None):
-        raise exc.Unsupported("Direct ServerPort.create_or_update call is unsupported.")
+        refs = self._refs_from_ser(conn)
+        sdk_params = self._to_sdk_params(refs)
+
+        # creation using neutron, NOTE: add update method
+        sdk_port = conn.network.create_port(**sdk_params)
+        return {'port_id': sdk_port['id']}
```

Comment:
The 'port_id' could be the culprit, see the docs re the 'networks' param for create_server: 

https://docs.openstack.org/api-ref/compute/?expanded=create-server-detail#create-server